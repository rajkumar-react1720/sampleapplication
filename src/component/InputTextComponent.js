import React from 'react';
import { InputText } from 'primereact/inputtext';


export function inputTextComponent(){
    return(
        <div>
            <InputText id='inputtextStyle' label='Name' placeholder='Please Enter Name...'>
          </InputText>
        </div>
    )
}

export default inputTextComponent;