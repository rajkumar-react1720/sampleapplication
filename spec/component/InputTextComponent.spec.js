import React from 'react';
import { shallow } from 'enzyme';
import InputTextComponent from '../../src/component/InputTextComponent';



describe('Rendering App Component without crash', () => {
    let inputTextComponent;
    beforeEach(() => {
        inputTextComponent = shallow(
          <InputTextComponent/>
        );
    });

    describe('TEST FEILD', () => {
        it('Render text feild', () => {
            //expect(inputTextComponent.find('#inputtextStyle').props().label).toEqual('Name');
            expect(inputTextComponent.find('#inputtextStyle').props().placeholder).toEqual('Please Enter Name...');
            expect(inputTextComponent).toMatchSnapshot();
        })
        
    })
})