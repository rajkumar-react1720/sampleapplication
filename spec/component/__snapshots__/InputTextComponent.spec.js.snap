// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering App Component without crash TEST FEILD Render text feild 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <inputTextComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <InputText
        id="inputtextStyle"
        keyfilter={null}
        label="Name"
        onInput={null}
        onKeyPress={null}
        placeholder="Please Enter Name..."
        tooltip={null}
        tooltipOptions={null}
        validateOnly={false}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "id": "inputtextStyle",
        "keyfilter": null,
        "label": "Name",
        "onInput": null,
        "onKeyPress": null,
        "placeholder": "Please Enter Name...",
        "tooltip": null,
        "tooltipOptions": null,
        "validateOnly": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <InputText
          id="inputtextStyle"
          keyfilter={null}
          label="Name"
          onInput={null}
          onKeyPress={null}
          placeholder="Please Enter Name..."
          tooltip={null}
          tooltipOptions={null}
          validateOnly={false}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "inputtextStyle",
          "keyfilter": null,
          "label": "Name",
          "onInput": null,
          "onKeyPress": null,
          "placeholder": "Please Enter Name...",
          "tooltip": null,
          "tooltipOptions": null,
          "validateOnly": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
